src/table_factory.h

```diff
@@ -45,6 +48,8 @@ class TitanTableFactory : public TableFactory {
 
   void* GetOptions() override { return base_factory_->GetOptions(); }
 
+  TitanCFOptions* GetCFOptions() { return &cf_options_; }
```

Comment:
Should we make the change reflected in cf_options_ whenever used? Since passing a bool is not scalable, and the `GetPrintableTableOptions` will return stale data too.

Reply:
Let's add two new structs, `ImmutableTitanCFOptions` and `MutableTitanOptions`, which contain non-changeable/changeable options from `TitanCFOptions`.